spec/puppet-server-lib/puppet/jvm/config_spec.rb

```diff
@@ -4,16 +4,46 @@
 
 describe Puppet::Server::Config do
   context "SSL context" do
-    it "is configured with :localcacert and not :cacert" do
+    before :each do
       Puppet[:hostcert] = "spec/fixtures/localhost-cert.pem"
       Puppet[:hostprivkey] = "spec/fixtures/localhost-privkey.pem"
-      Puppet[:cacert] = "SHOULD/NOT/BE/USED"
       Puppet[:localcacert] = "spec/fixtures/ca-cert.pem"
+    end
+
+    it "is configured with :localcacert and not :cacert" do
+      Puppet[:cacert] = "SHOULD/NOT/BE/USED"
+
       begin
-        Puppet::Server::Config.ssl_context
+        ssl_context = Puppet::Server::Config.ssl_context
       rescue Java::JavaIO::FileNotFoundException
         fail 'SSL context configured with :cacert'
       end
     end
+
+    it "ssl_context and puppet_only_ssl_context are the same SSLContext" do
+      ssl_context = Puppet::Server::Config.ssl_context
+      puppet_only_ssl_context = Puppet::Server::Config.puppet_only_ssl_context
+
+      expect(ssl_context).to be_a_kind_of(Java::JavaxNetSsl::SSLContext)
+      expect(ssl_context).to eq(puppet_only_ssl_context)
+    end
+
+    it "loads the puppet certs and any existing configured system certs w/o error" do
+      Puppet[:ssl_trust_store] = "spec/fixtures/puppet-cacerts"
+      begin
+        ssl_context = Puppet::Server::Config.puppet_and_system_ssl_context
+        expect(ssl_context).to be_a_kind_of(Java::JavaxNetSsl::SSLContext)
+      rescue => e
+        fail 'Could not initialize puppet_and_system_ssl_context'
+      end
+    end
+
+    it "warns if :ssl_trust_store is set but not readable" do
+      Puppet::Server::Config.instance_variable_set(:@puppet_and_system_ssl_context, nil)
```

Comment:
We save the result of calling `puppet_and_system_ssl_context` in this class level instance variable, so we need to set it back to nil between examples. Maybe I should move this and the equivalent changes to a before block?

Reply:
ahh, yeah, I think putting it in a `before` or `after` block would make sense. This makes it looks like something particular to this test, rather than a cleanup from the previous one.